File: solutions/csharp/interest-is-interesting/1/InterestIsInteresting.cs
using System;

static class SavingsAccount
{
    public static float InterestRate(decimal balance)
    {
        float interest = 2.475f;
        if (balance < 0.0m)
        {
            interest = 3.213f;
        } else if (balance < 1000.0m)
        {
            interest = 0.5f;
        } else if (balance < 5000)
        {
            interest = 1.621f;
        }
        return interest;
    }

    public static decimal Interest(decimal balance) => (balance * (decimal)(InterestRate(balance) / 100));

    public static decimal AnnualBalanceUpdate(decimal balance) => (balance + Interest(balance));

    public static int YearsBeforeDesiredBalance(decimal balance, decimal targetBalance)
    {
        int years = 0;
        if (balance >= targetBalance)
        {
            return years;
        }
        do
        {
            balance = AnnualBalanceUpdate(balance);
            ++years;
        } while (balance < targetBalance);
        return years;
    }
}

File: solutions/clojure/interest-is-interesting/1/src/interest_is_interesting.clj
(ns interest-is-interesting)

(defn interest-rate
  "Returns the interest rate based on the specified balance."
  [balance]
  (cond
    (neg? balance)   -3.213
    (< balance 1000)  0.5
    (< balance 5000)  1.621
    :else             2.475
    )
  )

(defn interest-multiplier
  "Returns the decimal value to multiply by"
  [interest]
  (/ (if (neg? interest)
    (+' -100.0 interest)
    (+' 100.0 interest)
    ) 100.0)
  )

(defn annual-balance-update
  "Returns the annual balance update, taking into account the interest rate."
  [balance]
  (*' balance (bigdec(abs (interest-multiplier (interest-rate balance)))))
)

(defn amount-to-donate
  "Returns how much money to donate based on the balance and the tax-free percentage."
  [balance tax-free-percentage]
  (if (pos? balance) (int (/ (*' 2 tax-free-percentage balance) 100)) 0)
  )